crates/subspace-networking/src/node_runner.rs

```diff
@@ -164,6 +176,62 @@ impl NodeRunner {
                             );
                         }
                     },
+                    KademliaEvent::OutboundQueryCompleted {
+                        id,
+                        result: QueryResult::GetRecord(results),
+                        ..
+                    } => {
+                        if let Some(QueryResultSender::GetValue { sender }) =
+                            self.query_id_receivers.remove(&id)
+                        {
```

Comment:
Why?

Reply:
I mean, as it not reported. I thought that it is not possible by algorithm logic.

Reply:
It is not implemented (not yet at least), but the key can be removed from the map in case requester no longer cares about the result, so I thought it doesn't matter too much why it is not there, we only care when it is. I'll think about better handling for this later.